apps/prairielearn/src/lib/editors.ts

```diff
@@ -950,11 +955,21 @@ export class CourseInstanceCopyEditor extends Editor {
     const short_name = names.shortName;
     const courseInstancePath = path.join(courseInstancesPath, short_name);
 
-    const fromPath = path.join(courseInstancesPath, this.course_instance.short_name);
     const toPath = courseInstancePath;
 
-    debug(`Copy template\n from ${fromPath}\n to ${toPath}`);
-    await fs.copy(fromPath, toPath, { overwrite: false, errorOnExist: true });
+    debug(`Copy course instance\n from ${this.from_path}\n to ${toPath}`);
+    await fs.copy(this.from_path, toPath, { overwrite: false, errorOnExist: true });
+
+    if (!idsEqual(this.course.id, this.from_course.id)) {
+      if (!this.from_course.sharing_name) {
+        throw new AugmentedError("Can't copy from course which hasn't declared a sharing name", {});
+      }
+      // Update the infoAssessment.json files to include the course sharing name for each question
+      await updateInfoAssessmentFilesForInstanceCopy(
+        courseInstancePath,
+        this.from_course.sharing_name,
+      );
```

Comment:
I originally panicked reading this thinking we were modifying the course instance in place. But we're not because of the weird file transfer machinery, right? Maybe add a comment explaining that this is safe because we're operating on a copy of the course instance files.

Reply:
done in 235a27b2f4